src/main/java/com/example/bank/registration/ClientService.java

```diff
@@ -0,0 +1,76 @@
+package com.example.bank.registration;
+
+import com.example.bank.exception.ClientNotFoundException;
+import com.example.bank.registration.jpa.Client;
+import com.example.bank.registration.jpa.ClientRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.time.Period;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ClientService {
+    private final ClientRepository clientRepository;
+    private final String clientLessThan18YearsOldMessage = "Client has to be adult.";
+
+    Client registerClient(ClientRequest clientRequest) {
+        log.info("Start client's registration.");
+        int years = Period.between(clientRequest.birthDate(), LocalDate.now())
+                .getYears();
+        if (years >= 18) {
```

Comment:
to 18 bym zamienil na stala, jest uzywana tu w jednym miejscu ale bym to raczej wydzielil do stalej do klasy.
Bo musimy myslec globalnie, jezeli np Twoj system bedzie uzywany w USA a tam pelnoletnosc to 21 lat np. 
Wiec mozna by to opzniej nawet do app.yml wrzuci c

Reply:
Od razu dodałem do pliku yaml'a.